community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Read.java

```diff
@@ -73,6 +73,13 @@ void nodeIndexSeek( IndexReference index, NodeValueIndexCursor cursor, IndexOrde
      */
     void singleNode( long reference, NodeCursor cursor );
 
+    /**
+     * Checks if a node exists in the database
+     * @param id The id of the node to check
+     * @return <tt>true</tt> if the node exists, otherwise <tt>false</tt>
+     */
+    boolean nodeExists( long id );
```

Comment:
A little test of this would be nice, as it lies in the API.

Reply:
Added tests